deepreg/callback.py

```diff
@@ -1,38 +1,39 @@
 import tensorflow as tf
 
 
-def build_callbacks(
-    model, dataset, log_dir: str, histogram_freq: int, save_period: int
-) -> list:
+def build_checkpoint_manager_callback(
+    model, dataset, log_dir: str, save_period: int, ckpt_path: str
+):
     """
     Function to prepare callbacks for training.
 
     :param log_dir: directory of logs
     :param histogram_freq: save the histogram every X epochs
     :param save_period: save the checkpoint every X epochs
+    :param ckpt_path: path to restore ckpt
     :return: a list of callbacks
     """
-    tensorboard_callback = tf.keras.callbacks.TensorBoard(
-        log_dir=log_dir, histogram_freq=histogram_freq
-    )
-
     # fit the model for 1 step to initialise optimiser arguments as trackable Variables
     model.fit(
         x=dataset,
         steps_per_epoch=1,
         epochs=1,
     )
-
     checkpoint_manager_callback = CheckpointManagerCallback(
         model, log_dir + "/save", period=save_period
     )
-    return [tensorboard_callback, checkpoint_manager_callback]
+    if ckpt_path:
+        initial_epoch = int(ckpt_path.split("-")[-1])
+        checkpoint_manager_callback.restore(ckpt_path)
+    else:
+        initial_epoch = 0
+    return checkpoint_manager_callback, initial_epoch
 
 
 def restore_model(callbacks, ckpt_path):
```

Comment:
remove this func as i think its no longer used

Reply:
done